olp-cpp-sdk-core/include/olp/core/thread/TaskScheduler.h

```diff
@@ -138,5 +138,14 @@ class CORE_API TaskScheduler {
   }
 };
 
+/**
+ * @brief Common function used to add a lambda function and  schedule this to a
+ * task scheduler.
+ * @param task_scheduler Task scheduler instance.
+ * @param func Function that will be executed.
+ */
+void ExecuteOrSchedule(const std::shared_ptr<TaskScheduler>& scheduler,
```

Comment:
Unit tests missing.

Reply:
added two tests to ThreadPoolTaskSchedulerTest